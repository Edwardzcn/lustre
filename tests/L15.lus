node nats() returns (n : int)
let
  n = 0 -> 1 + pre n;
tel

node main(c : bool) returns (n : int)
  var x : int when c;
let
  x = callWhen(c,nats());
  n = currentWith(17,x);
tel

