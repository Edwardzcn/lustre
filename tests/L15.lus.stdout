PhaseToCore
===========

node (c : bool)
  returns (n)
  assumes ()
  shows ()
let
  valof_pre_x : int when c
    = pre x
   
  x : int when c
    = 0 -> Add(1, valof_pre_x)
   
  valof_current : int
    = current x
   
  valof_pre_n : int
    = pre n
   
  valof_arr : int
    = 17 -> valof_pre_n
   
  n : int
    = ITE(c, valof_current, valof_arr)
tel

--- Step 1 ---
c : bool  = False
n = 17
--- Step 2 ---
c : bool  = False
n = 17
--- Step 3 ---
c : bool  = True
n = 0
--- Step 4 ---
c : bool  = False
n = 0
--- Step 5 ---
c : bool  = False
n = 0
--- Step 6 ---
c : bool  = True
n = 1
--- Step 7 ---
c : bool  = True
n = 2
--- Step 8 ---
c : bool  = False
n = 2
--- Step 9 ---
c : bool  = False
n = 2
--- Step 10 ---
c : bool  = (EOF)
